src/speex_resample.rs

```diff
@@ -636,9 +636,9 @@ impl SpeexResamplerState {
      */
     pub fn reset_mem(&mut self) {
         let nb_channels = self.nb_channels as usize;
-        self.last_sample = vec![0; nb_channels];
-        self.magic_samples = vec![0; nb_channels];
-        self.samp_frac_num = vec![0; nb_channels];
+        self.last_sample.clear(); self.last_sample.resize(nb_channels, 0);
+        self.magic_samples.clear(); self.magic_samples.resize(nb_channels, 0);
+        self.samp_frac_num.clear(); self.samp_frac_num.resize(nb_channels, 0);
 
```

Comment:
Could you use the same approach of `self.mem`, please? It's much clearer, thanks!